### ISSUE TITLE
Incorrectly reporting on large numbers within a range

### ISSUE THREAD

**Issue by andrewtimberlake** [rrrene/credo#95]
When running `mix credo -a` against

```elixir
def test do
  10_000..20_000
end
```

Credo suggests using an underscore for large numbers.

```
Checking 1 source file ...

  Code Readability
┃
┃ [R] ↗ Large numbers should be written with underscores: 10_000
┃       lib/credo_test.ex:5:5 (CredoTest.test)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.1 seconds (0.00s to load, 0.1s running checks)
2 mods/funs, found 1 code readability issue.

Showing priority issues: ↑ ↗ →  (use `--strict` to show all issues, `--help` for options).
```

I have uploaded a test repo at https://github.com/andrewtimberlake/credo_test
The file being checked is https://github.com/andrewtimberlake/credo_test/blob/master/lib/credo_test.ex

**Comment by rrrene** [comment]
Hi Andrew,

this was fixed and will be part of `v0.4.0`! Thanks for reporting the issue! :+1:

**Comment by p-kraszewski** [comment]
Still happens:
```
defmacro oid_ansi_x9_62, do: quote do: {1,2,840,10_045}
```
is reported.

Credo 0.4.5 @ Elixir 1.3.0 @ Erlang 19.0 @ Ubuntu 16.04

**Comment by rrrene** [comment]
Crap.

**Comment by rrrene** [comment]
And by "Crap." I mean: Thx for reporting this :wink:

**Comment by p-kraszewski** [comment]
Yes, of course, I got the right idea already on your 1st answer.
I'll drop here a minimal (non)working example tomorrow.

**Comment by aweiker** [comment]
I just ran across this issue with 0.4.5 when using this format.
```elixir
      bounds: [
        0, 1, 2, 5, 10, 20, 30, 65, 85,
        100, 200, 400, 800,
        1_000, 2_000, 4_000, 8_000, 16_000]
```

**Comment by barruumrex** [comment]
Ran into this issue in 0.4.6 with `Checking 1 source file ...

  BridgeApi
┃
┃   [R] Category: readability
┃    ↗  Priority: medium
┃
┃       Large numbers should be written with underscores: 10_000
┃       lib/bridge_api.ex:11:63 (BridgeApi.start)
┃
┃    __ CODE IN QUESTION
┃
┃     9
┃    10     #Install core-api dependencies
┃    11     :fuse.install(:core_api, {{:standard, 3, 10_000}, {:reset, 10_000}})
┃                                                                     ^^^^^^
`

**Comment by rrrene** [comment]
@barruumrex which Elixir version are you using?

**Comment by barruumrex** [comment]
Oh. Of course. 
Elixir: 1.3.1
Erlang: 18.3
Operating System: osx

**Comment by rrrene** [comment]
If its no trouble, could you test with Elixir `1.3.2`? We deployed a fix in Credo `0.4.6` which unfortunately needs info the tokenizer only provides starting with `1.3.2` ...

**Comment by barruumrex** [comment]
Yeah. Bumping up to 1.3.2 fixes it. Thanks.

**Comment by cpjolicoeur** [comment]
We are using elixir 1.3.2 and credo 0.4.7 and we still see this issue.  Numbers such as `500_000` and `2_000_000` are tripping the `Large numbers should be written with underscores` readability rule when they shouldn't be since they already have underscores.

**Comment by rrrene** [comment]
@cpjolicoeur I tested with a naive range (`500_000..2_000_000`) and it di not yield an issue. Could you provide a code snippet I can add to the test suite?

**Comment by cpjolicoeur** [comment]
@rrrene Sorry, mine is slightly different and not a range, just plain numbers.  But here is a snippet.

```
  @budgets %{
    "budget1": 100_000,
    "budget2": 200_000,
    "budget3": 300_000,
    "budget4": 500_000,
    "budget5": 1_000_000,
    "budget6": 2_000_000
  }
```

**Comment by rrrene** [comment]
Hi, re-reading your bug report, I'm not sure why I assumed a range. Too much (or too little) coffee, I guess ;-)

The test is breaking now, so I will work on a fix :+1:
